addons/web/models/models.py

```diff
@@ -616,9 +634,20 @@ def _web_read_group_format(
         for dict_group in result:
             dict_group['__extra_domain'] = AND(dict_group.pop('__extra_domains'))
 
+        # Link the monetary columns to their currency columm
+        currency_column_mapping = {}
+        column_values = list(zip(*groups))[len(groupby):]
```

Comment:
`list(zip(*groups))` will duplicate all values (even if no monetary is used). It is not a big deal but we should try to make this method the most efficient in term of memory and speed.

See my next comment for a solution.